hnne/projector.py

```diff
@@ -150,7 +177,8 @@ def fit(
             self,
             X: np.ndarray,
             y: np.ndarray = None,
-            dim: int = 2,
+            dim: Union[Optional[int], None] = None,
```

Comment:
The type can be simplified to `Optional[int]`: https://docs.python.org/3/library/typing.html#typing.Optional